components/automate-cli/cmd/chef-automate/ha_node_delete_aws.go

```diff
@@ -189,33 +189,64 @@ func (dna *DeleteNodeAWSImpl) promptUserConfirmation() (bool, error) {
 	return dna.writer.Confirm("This will delete the above node from your existing setup. It might take a while. Are you sure you want to continue?")
 }
 
+func (dna *DeleteNodeAWSImpl) saveConfigToBastion(infra *AutomateHAInfraDetails) error {
+	nodeObjects := getNodeObjectsToFetchConfigFromAllNodeTypes()
+	return executeCmdInAllNodeAndCaptureOutput(nodeObjects, true, AUTOMATE_HA_AUTOMATE_NODE_CONFIG_DIR, infra, dna.nodeUtils)
+}
+
+func (dna *DeleteNodeAWSImpl) syncConfigToAllNodes(infra *AutomateHAInfraDetails) error {
+	nodeObjects := getNodeObjectsToPatchWorkspaceConfigToAllNodes()
+	return executeCmdInAllNodeAndCaptureOutput(nodeObjects, false, "", infra, dna.nodeUtils)
+}
+
 func (dna *DeleteNodeAWSImpl) runDeploy() error {
-	err := dna.runRemoveNodeFromAws()
+	infra, _, err := dna.nodeUtils.getHaInfraDetails()
 	if err != nil {
 		return err
 	}
+	err = dna.saveConfigToBastion(infra)
+	if err != nil {
+		return errors.Wrap(err, "error saving node configuration to bastion")
+	}
+	err = dna.runRemoveNodeFromAws()
+	if err != nil {
+		return errors.Wrap(err, "error removing node from AWS")
+	}
 
 	err = dna.nodeUtils.moveAWSAutoTfvarsFile(dna.terraformPath)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "error moving aws.auto.tfvars file")
 	}
 
 	err = dna.nodeUtils.modifyTfArchFile(dna.terraformPath)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "error modifying tf_arch file")
 	}
 	err = dna.nodeUtils.writeHAConfigFiles(awsA2harbTemplate, dna.config)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "error writing HA config.toml in workspace directory")
 	}
 
 	argsdeploy := []string{"-y"}
 	err = dna.nodeUtils.executeAutomateClusterCtlCommandAsync("provision", argsdeploy, provisionInfraHelpDocs)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "error while provisioning resources")
+	}
+
+	err = dna.nodeUtils.executeAutomateClusterCtlCommandAsync("deploy", argsdeploy, upgradeHaHelpDoc)
+	if err != nil {
+		return errors.Wrap(err, "error while deploying architecture")
+	}
+	infra, _, _ = dna.nodeUtils.getHaInfraDetails()
+	syncErr := dna.syncConfigToAllNodes(infra)
+	if syncErr != nil {
+		err = errors.Wrapf(syncErr, "error syncing config to all nodes. \n%v", err)
```

Comment:
do nil check before combining err otherwise it might print nil

Reply:
Done